### PR TITLE
[pt1][quant] Add dequantize_linear for JIT pass

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #20656 [pt1][quant] int_repr for different quantized types&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15398134/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#20107 [pt1][quant] Add dequantize_linear for JIT pass**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D15202187/)

att

Differential Revision: [D15202187](https://our.internmc.facebook.com/intern/diff/D15202187/)

### CODE REVIEW

File: aten/src/ATen/Dispatch.h [hunk 1]
```diff
@@ -210,13 +217,13 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
...
-      AT_PRIVATE_CASE_TYPE(                                             \
-          at::ScalarType::QInt8, qint8, __VA_ARGS__)                    \
-      AT_PRIVATE_CASE_TYPE(                                             \
-          at::ScalarType::QUInt8, quint8, __VA_ARGS__)                  \
-      AT_PRIVATE_CASE_TYPE(                                             \
-          at::ScalarType::QInt32, qint32, __VA_ARGS__)                  \
+  switch (TYPE) {                                                       \
+      AT_QINT_PRIVATE_CASE_TYPE(                                        \
+          at::ScalarType::QInt8, qint8, int8_t, __VA_ARGS__)            \
+        AT_QINT_PRIVATE_CASE_TYPE(                                      \
```
Comment (nishantpdce) [inline aten/src/ATen/Dispatch.h]: alignment

**Comment by nishantpdce** [comment]
Please run clang-format on the .h/.cpp files

**Comment by dzhulgakov** [comment]
dumb question - why do you need this if you have as_quantized_tensor in the following diff? also, how is it going to be used in the  JIT pass, just  curious?

**Comment by jerryzh168** [comment]
@dzhulgakov in JIT pass, we want explicit scale, zero_point argument in dequantize so that we don't need to traverse back to find these argument in quantize_linear. We don't expect this function will be used by user, this is for JIT pass only.

**Pull request closed without merging**